### DISCUSSION TITLE
Client secret could not be decrypted while defined by Recipe

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#8773] [Q&A] [answered]
The client Id secret defined by recipe  recive error in log:

_OrchardCore.OpenId.Configuration.OpenIdClientConfiguration|ERROR|The client secret could not be decrypted. It may have been encrypted using a different key._ 

```
    {
      "name": "OpenIdClientSettings",
      "Authority": "http://localhost:7001/auth/realms/app1/",
      "DisplayName": "My Login",
      "ClientId": "Local-Manager",
      "CallbackPath": "/signin-oidc",
      "SignedOutCallbackPath": "/signout-callback-oidc",
      "Scopes": "email profile",
      "ResponseMode": "form_post",
      "ResponseType": "code id_token",
      "ClientSecret": "6ef57fa1-9118-4455-a226-9b45h7eb93cd"
    },
```

**Comment by deanmarcussen** [comment, marked as answer]
Yes, you will have to resave it after importing due to the change in data protection keys.

See : https://github.com/OrchardCMS/OrchardCore/pull/7891
https://github.com/OrchardCMS/OrchardCore/issues/5558

**Reply by vitalybrandes** [reply]
ohhh :-(
Am i right that the secrets module will fix this it the future?
is there any workaround?

**Reply by vitalybrandes** [reply]
> 
> 
> Yes, you will have to resave it after importing due to the change in data protection keys.
> 
> See : #7891
> #5558

@deanmarcussen 
do you have any estimation when this secrets module will be ready?
